### PR TITLE
syncer: add gtid

### PR DESCRIPTION
**Pull request by GregoryIan**
@andelf @iamxy PTAL

### CODE REVIEW

**Comment by iamxy** [comment]
@qiuyesuifeng @siddontang @nolouch PTAL

**Comment by GregoryIan** [comment]
currently only support  to configure gtid  in the meta file.  if add a parameter, the priority of cmd parameters will affect restart operation.

**Comment by andelf** [comment]
LGTM, update README and metrics

**Comment by iamxy** [comment]
LGTM

File: syncer/syncer.go [hunk 1]
```diff
@@ -760,6 +772,31 @@ func (s *Syncer) printStatus() {
 	}
 }
 
+func (s *Syncer) getBinlogStreamer() (*replication.BinlogStreamer, error) {
+	gtid := s.meta.GTID()
+
+	if gtid != "" {
+		var f func(string) (mysql.GTIDSet, error)
```
Comment (holys) [inline syncer/syncer.go]: Why not use [ParseGTIDSet](https://github.com/siddontang/go-mysql/blob/master/mysql/gtid.go#L16) ?

**Pull request merged**